### PR TITLE
.Net: Adding chat system prompt to config file schema

### LINKED ISSUE #644: Is there any way to define the system/user/assistant messages in the prompt file with the `OpenAIChatCompletion`, but not call the `add_system_message`/`add_user_message`/`add_assistant_message` in code?
Is there any way to define the system/user/assistant messages in the prompt file with the `OpenAIChatCompletion`, but not call the `add_system_message`/`add_user_message`/`add_assistant_message` in code?

### LINKED ISSUE #2068: How to change system message when use skprompt.txt with Chat Completion API
When use semantic function which from skprompt.txt file, how to change system message?

I noticed that the default system message always is: You are an AI assistant that helps people find information.

### PR DESCRIPTION
**Pull request by RogerBarreto**
### Motivation and Context

Adding support to set the chat system message using `config.json` with Semantic Functions.

```
{
  ...
  "completion": {
    ...
    "chat_system_prompt": "My custom system chat prompt"
  }
}
```

Resolves #644 
Closes #644 
Resolves #2068 
Closes #2068 

### Description

Allows configuration files to specify what will be the system chat message used while requesting from LLMs.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Comment by lemillermicrosoft** [comment]
We should make sure this is done in Python as well (either open a different issue or don't close the linked upon completion.

**Comment by lemillermicrosoft** [comment]
Should we also add support to `CreateSemanticFunction` extensions?

File: dotnet/src/SemanticKernel.Abstractions/AI/TextCompletion/CompleteRequestSettings.cs [hunk 1]
```diff
@@ -81,5 +81,12 @@ public static CompleteRequestSettings FromCompletionConfig(PromptTemplateConfig.
             MaxTokens = config.MaxTokens,
             StopSequences = config.StopSequences,
         };
+
+        if (config.ChatSystemPrompt is not null)
```
Comment (SergeyMenshykh) [inline dotnet/src/SemanticKernel.Abstractions/AI/TextCompletion/CompleteRequestSettings.cs:85]: What if the `ChatSystemPrompt` property is empty? should it be added?

**Review by SergeyMenshykh — APPROVED** [review]

File: dotnet/src/SemanticKernel.Abstractions/AI/TextCompletion/CompleteRequestSettings.cs (hunk 1 above)
Comment (RogerBarreto) [reply dotnet/src/SemanticKernel.Abstractions/AI/TextCompletion/CompleteRequestSettings.cs:85]: Good point, going to change it to `string.IsNullOrWhitespace()`

File: dotnet/src/SemanticKernel.UnitTests/KernelTests.cs [hunk 2]
```diff
@@ -125,6 +126,34 @@ public void ItImportsSkillsNotCaseSensitive()
         Assert.True(skill.ContainsKey("GETANYVALUE"));
     }
 
+    [Theory]
+    [InlineData(null, "Assistant is a large language model.")]
```
Comment (dmytrostruk) [inline dotnet/src/SemanticKernel.UnitTests/KernelTests.cs:130]: Where did this `Assistant is a large language model.` message come from?

Comment (RogerBarreto) [reply dotnet/src/SemanticKernel.UnitTests/KernelTests.cs:130]: https://github.com/microsoft/semantic-kernel/blob/c8a7c6d134fff68e59dd8cc191e9a2fdc1d0937a/dotnet/src/SemanticKernel.Abstractions/AI/TextCompletion/CompleteRequestSettings.cs#L61

**Review by dmytrostruk — APPROVED** [review]

**Comment by pablontiv** [comment]
Is this already available in Semantic Kernel Tools?

**Comment by sandeepvootoori** [comment]
Is there a reason why this was implemented in the config.json? putting it in config.json makes it difficult to hydrate the system message with grounding data while using planners. Any plans in the future to make this part of skprompt.txt?

@thegovind

**Pull request merged**